sdks/python/apache_beam/pipeline.py

```diff
@@ -314,7 +314,15 @@ def _replace_if_needed(self, original_transform_node):
             new_output.element_type = None
```

Comment:
right below new_output assignment, please assert that it is one of the supported types:
```
assert isinstance(result, (dict, pvalue.PValue, pvalue.DoOutputsTuple))
```

Reply:
Added the assert